app/code/Magento/NewsletterGraphQl/Model/Resolver/SubscribeEmailToNewsletter.php

```diff
@@ -0,0 +1,237 @@
+            $storeId = (int)$context->getExtensionAttributes()->getStore()->getId();
+            $websiteId = (int)$context->getExtensionAttributes()->getStore()->getWebsiteId();
+
+            $this->validateEmailFormat($email);
+            $this->validateGuestSubscription($context);
+            $this->validateEmailAvailable($email, $currentUserId, $websiteId);
+            $this->validateAlreadySubscribed($email, $websiteId);
+
+            $subscriber = $this->isCustomerSubscription($email, $currentUserId)
+                ? $this->subscriptionManager->subscribeCustomer($currentUserId, $storeId)
+                : $this->subscriptionManager->subscribe($email, $storeId);
+
+            $status = $this->enumLookup->getEnumValueFromField(
+                'SubscriptionStatusesEnum',
+                (string)$subscriber->getSubscriberStatus()
+            );
+        } catch (LocalizedException $e) {
+            throw new GraphQlInputException(__($e->getMessage()));
+        }
+
+        return [
+            'status' => $status
+        ];
+    }
+
+    /**
+     * Validate the format of the email address
+     *
+     * @param string $email
+     * @throws GraphQlInputException
+     */
+    private function validateEmailFormat(string $email): void
+    {
+        if (!$this->emailValidator->isValid($email)) {
+            throw new GraphQlInputException(__('Enter a valid email address.'));
+        }
+    }
+
+    /**
+     * Validate if a guest user can be subscribed to a newsletter.
+     *
+     * @param ContextInterface $context
+     * @throws GraphQlInputException
+     */
+    private function validateGuestSubscription(ContextInterface $context): void
+    {
+        if (false === $context->getExtensionAttributes()->getIsCustomer()
+            && !$this->scopeConfig->getValue(
+                Subscriber::XML_PATH_ALLOW_GUEST_SUBSCRIBE_FLAG,
+                ScopeInterface::SCOPE_STORE
+            )
+        ) {
+            throw new GraphQlInputException(
+                __('Guests can not subscribe to the newsletter. You must create an account to subscribe.')
+            );
+        }
+    }
+
+    /**
+     * Validates that the email address isn't being used by a different account.
+     *
+     * @param string $email
+     * @param int $currentUserId
+     * @param int $websiteId
+     * @throws GraphQlInputException
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    private function validateEmailAvailable(string $email, int $currentUserId, int $websiteId): void
+    {
+        if ($currentUserId > 0) {
+            $customer = $this->customerRepository->getById($currentUserId);
+
+            if ($customer->getEmail() != $email
+                && !$this->customerAccountManagement->isEmailAvailable($email, $websiteId)) {
+                throw new GraphQlInputException(
+                    __('This email address is already assigned to another user.')
```

Comment:
Such information should not be exposed.

Reply:
I got this part from the [original method](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Newsletter/Controller/Subscriber/NewAction.php#L99).

Reply:
**done**